api/ccapi/include/layer.h

```diff
@@ -107,6 +107,9 @@ class Layer {
    *            31. in_dim : int ( input dimension for embedding layer )
    *            32. out_dim : int ( output dimesion for embedding layer )
    *            33. in_length : int ( input length for embedding layer )
+   *            34. recurrent_activation :  string (type) - lstm
+   *            35. dist_layer : string (type) - layer name to be distributed
```

Comment:
```suggestion
   *            35. dist_layer : string (type) - layer type name to be distributed
```

Reply:
No.. I think it must be layer name. not type name. This could be defined by user.

Reply:
You might need to think a little more about this. Taking `layer name` for which time distribution is to be done would require the graph to add a layer, then give that layer to `time distributed` layer and replace it inside the graph.
From the API perspective, adding `time distributed` layer to the graph removes the original layer from the graph which is also a bit weird. 

As a workaround, you might want to consider a design, where time_dist is not a layer but rather a wrapper for the layer. When time distributed is set (like flatten), then it adds an additional loop for each kind of layer. This might come out as a cleaner way and have less obstruction on the graph and less confusion for the user.

Reply:
I see, understood :)

Reply:
I wonder how it would be used in the ini. If input_layers is not specified, it is automatically connects to a previous layer.

```ini
[fclayer]
type=fc

[convlayer]
type=conv
// this will implicitly have fclayer as an input in the graph as well.

[dist_layer]
dist_layer=fclayer
input_layers=fc
```

Reply:
yes. it is automatically connects the previous layer and dist_layer inside.